lib/salus/report.rb

```diff
@@ -301,22 +313,49 @@ def write_report_to_file(report_file_path, report_string)
             "Cannot write file #{report_file_path} - #{e.class}: #{e.message}"
     end
 
-    def send_report(remote_uri, data, format)
-      response = Faraday.post do |req|
-        req.url remote_uri
-        req.headers['Content-Type'] = CONTENT_TYPE_FOR_FORMAT[format]
-        req.headers['X-Scanner'] = x_scanner_type(format)
-        req.body = data
+    def report_headers_h(headers, headers_env_var, format)
+      header_hash = {}
+      header_hash['Content-Type'] = CONTENT_TYPE_FOR_FORMAT[format]
+      header_hash['X-Scanner'] = x_scanner_type(format)
+
+      # Iterate through specified headers and assign values
+      headers.each do |field, value|
+        header_hash[field] = value
+      end
+
+      headers_env_var.each do |field, value|
+        header_hash[field] = ENV[value] || ''
       end
 
+      header_hash
+    end
+
+    def send_report(config, remote_uri, data, format)
+      conn = Faraday.new(
+        url: remote_uri,
+        headers: report_headers_h(config['headers'] || {},
+                                  config['headers_env_var'] || {},
+                                  format)
+      )
+
+      response = if config&.key?('put')
+                   conn.put do |req|
+                     req.body = data
+                   end
+                 else
+                   conn.post do |req|
+                     req.body = data
+                   end
+                 end
+
       unless response.success?
         raise ExportReportError,
-              "POST of Salus report to #{remote_uri} had response status #{response.status}."
+              "Salus report to #{remote_uri} had response status #{response.status}."
       end
     end
 
     def report_body_hash(config, data)
-      return data unless config&.key?('post')
+      return data unless config&.key?('post') && !config['post'].nil?
```

Comment:
config['key'].present?

Salus has activesupport in the gemfile so you can take advantage of some of that syntactical sugar